example_impl/services/socket.go

```diff
@@ -0,0 +1,6 @@
+package services
+
+type Socket struct {
```

Comment:
is this generic struct something thats going to be repeated for every impl that uses the Go client?

Reply:
Nope! It really depends on what your definition of `Service` is. E.g. we _could_ have not had a `GeckoClient` class, and just made the `Service` do it. I did this because it was brainless, and a good example